File: 2024 styczeń/INF.04-02-24.01-SG & INF.04-01-24.01-SD/Aplikacja Konsolowa/C++/main.cpp
#include <iostream>
#include <algorithm>

using namespace std;

class StringTools
{
public:
    static int vowelCounter(string text)
    {
        const string vowels = "aąeęiouóyAĄEĘIOUÓY";

        int vowelCount = 0;

        for (char ch : text)
        {
            if (vowels.find(ch) != string::npos)
            {
                vowelCount++;
            }
        }

        return vowelCount;
    }

    static string repetitionDeleter(string text)
    {
        string result;

        for (int i = 0; i < text.size(); i++)
        {
            if (i == 0 || text[i] != text[i - 1])
            {
                result += text[i];
            }
        }

        return result;
    }
};

int main()
{
    string text;

    cout << "Podaj napis: ";
    getline(cin, text);


    StringTools newStringTools;

    int vowelCount = newStringTools.vowelCounter(text);
    string newText = newStringTools.repetitionDeleter(text);

    cout << "Liczba samogłosek: " << vowelCount << endl;
    cout << "Napis po usunięciu powtórzeń: " << newText << endl;

    return 0;
}

File: 2024 styczeń/INF.04-02-24.01-SG & INF.04-01-24.01-SD/Aplikacja Konsolowa/Python/main.py
"""
    klasa: StringTools
    opis: klasa representuje zestaw narzędzi do napisów
    metody: vowelCounter - zwraca ilość samogłosek w napisie
            repetitionDeleter - zwraca napis z usuniętymi powtórzeniami znaków obok siebie
    autor: 00000000000
"""
class StringTools:
    @staticmethod
    def vowelCounter(text):
        VOWELS = "aąeęiouóyAĄEĘIOUÓY"
        vowelCount = 0
    
        for ch in text:
            if ch in VOWELS:
                vowelCount += 1
        
        return vowelCount
    
    @staticmethod
    def repetitionDeleter(text):
        result = ""
        
        for i in range(len(text)):
            if i == 0 or text[i] != text[i - 1]:
                result += text[i]
                
        return result

text = input("Podaj napis: ")

vowelCount = StringTools.vowelCounter(text)
newText = StringTools.repetitionDeleter(text)

print("Liczba samogłosek: ", vowelCount)
print("Napis po usunięciu powtórzeń: ", newText)